source/manual/retiring-a-gem.html.md

```diff
@@ -0,0 +1,46 @@
+---
+owner_slack: "#govuk-developers"
+title: Retire a gem
+section: Applications
+layout: manual_layout
+parent: "/manual.html"
+---
+
+## 1. Ensure the gem is not used by any apps
+
+Search for references to the gem in our codebase to see if any apps are still using it:
+
+[example](https://github.com/search?q=org%3Aalphagov+rack-logstasher+filename%3AGemfile+filename%3A*.gemspec&type=Code&ref=advsearch&l=&l=)
+
+If any repos are found to have the gem as a dependency, check whether the app is still actively using it by looking at the usage instructions in the gem's README and searching for that code in the repo:
+
+["use Rack::Logstasher::Logger"](https://github.com/alphagov/search-api/search?q=%22use+Rack%3A%3ALogstasher%3A%3ALogger%22)
+
+If the gem is being used you will either need to replace it with a different gem or your own code before you can archive it.
+
+When you have removed the gem from all apps, ensure the tests pass, monitor Sentry for alerts, etc.
+
+## 2. Update gem's README and archive it
+
+When the gem's repo is ready to be archived, push a commit explaining the reasons for archiving it.
+
+[Example](https://github.com/alphagov/govuk_taxonomy_helpers/pull/27)
+
+Go into the repository settings in GitHub, and
+[archive the repo](https://github.com/blog/2460-archiving-repositories).
+
+## 3. Remove references and update docs
+
+Do a [code search on GitHub][https://github.com/search?q=org%3Aalphagov+panopticon&type=Code] to find any references to the application
+and update or remove them.
+
+Mark the application as `retired` in [govuk-developer-docs][https://github.com/alphagov/govuk-developer-docs].
+
+## 4. Yank the gem
+
+Once the gem has been archived you may want to [yank it from Rubygems](https://guides.rubygems.org/removing-a-published-gem/) if it was published there. You can find the Rubygems credentials in [govuk-secrets](https://github.com/alphagov/govuk-secrets).
+
+## 5. Update the dependency management spreadsheet
+
+Remove the gem from the [GOV.UK dependency management spreadsheet](https://docs.google.com/spreadsheets/d/137KZhjctJ8qTKYPnq2QNVkyoIC6ok7KA2G1vErNC6Oo/edit#gid=1160366178)
```

Comment:
We can delete this I think - it would automatically happen next time we run "Update Data".